sign/dilithium/mode/mode.go

```diff
@@ -0,0 +1,66 @@
+// mode defines the interface for the various modes of the Dilithium
+// signature scheme modes.
+package mode
+
+import (
+	"io"
+)
+
+// PublicKey is a Dilithium public key.
+type PublicKey interface {
+	// Packs public key
+	Bytes() []byte
+}
+
+// PrivateKey is a Dilithium public key.
+type PrivateKey interface {
+	// Packs private key
+	Bytes() []byte
+}
+
+// Mode is a certain configuration of the Dilithium signature scheme.
+type Mode interface {
```

Comment:
I think this file should be on the top package, so anyone can see the functionality that a `Mode` provides.

Reply:
I would've liked that as well, but then there is a cyclic dependency.

Reply:
The top-level package needs to import the mode1, mode2, etc. subpackages. These subpackages themselves then would need to import the top-level package for the interfaces.